### PR TITLE
Update cardano-addresses to 3.1.0

### PR DESCRIPTION
**Pull request by hasufell**

### CODE REVIEW

**Review by KtorZ — APPROVED** [review]

**Comment by hasufell** [comment]
Updated to 3.1.0

File: lib/shelley/src/Cardano/Wallet/Shelley/Compatibility.hs [hunk 1]
```diff
@@ -232,6 +230,7 @@ import Shelley.Spec.Ledger.BaseTypes
 import Type.Reflection
     ( Typeable, typeRep )
 
+import qualified Cardano.Address.Style.Shelley as SH
```
Comment (rvl) [inline lib/shelley/src/Cardano/Wallet/Shelley/Compatibility.hs:233]: I would have preferred not changing it to use qualified imports. This module is called `...Shelley.Compatibility` so a prefix like `SH` (I assume it stands for Shelley?) is redundant.

**Review by rvl — APPROVED** [review]
:+1:

**Comment by rvl** [comment]
bors merge

**Pull request merged**